### PR TITLE
Update to the latest signalfx proto model that uses gogoproto

### PR DESCRIPTION
**Pull request by bogdandrutu**
* Improved 10K DPS from ~50 -> ~47.5 cpu usage
* Improved test coverage

### CODE REVIEW

File: exporter/signalfxexporter/metricdata_to_signalfxv2.go [hunk 1]
```diff
@@ -162,8 +161,7 @@ func metricDataToSignalFxV2(
 						numDroppedTimeSeries++
 						logger.Warn(
 							"Timeseries for distribution metric dropped",
-							zap.Error(err),
-							zap.String("metric", *sfxDataPoint.Metric))
```
Comment (jrcamp) [inline exporter/signalfxexporter/metricdata_to_signalfxv2.go:166]: Did you mean to drop "metric" log field?

File: exporter/signalfxexporter/metricdata_to_signalfxv2.go [hunk 2]
```diff
@@ -174,14 +172,13 @@ func metricDataToSignalFxV2(
 						numDroppedTimeSeries++
 						logger.Warn(
 							"Timeseries for summary metric dropped",
-							zap.Error(err),
-							zap.String("metric", *sfxDataPoint.Metric))
```
Comment (jrcamp) [inline exporter/signalfxexporter/metricdata_to_signalfxv2.go:178]: Same question here. I see you kept it below.

File: exporter/signalfxexporter/metricdata_to_signalfxv2.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/signalfxexporter/metricdata_to_signalfxv2.go:166]: The field is also in the error, feels like duplicate information for me.

File: exporter/signalfxexporter/metricdata_to_signalfxv2.go (hunk 2 above)
Comment (bogdandrutu) [reply exporter/signalfxexporter/metricdata_to_signalfxv2.go:178]: The field is also in the error, feels like duplicate information for me.

File: exporter/signalfxexporter/metricdata_to_signalfxv2.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/signalfxexporter/metricdata_to_signalfxv2.go:166]: Added back, will cleanup later.

File: exporter/signalfxexporter/metricdata_to_signalfxv2.go (hunk 2 above)
Comment (bogdandrutu) [reply exporter/signalfxexporter/metricdata_to_signalfxv2.go:178]: Added back, will cleanup later.

**Review by bogdandrutu — COMMENTED** [review]
PTAL

**Review by jrcamp — APPROVED** [review]

**Pull request merged**